### DISCUSSION TITLE
mlx on comfyUI

### DISCUSSION THREAD

**Discussion by azrahello** [comfyanonymous/ComfyUI#4559] [Ideas]
Hi everyone, I was wondering if there is a way to use or import something similar to https://github.com/filipstrand/mflux or https://github.com/ml-explore/mlx-examples/tree/main/stable_diffusion in ComfyUI. I don't know where to start, but if any developer needs it, I can provide access to my Mac Studio. CoreML converts models and makes them too "restrictive," but here there doesn't seem to be any conversion. It should be easier to implement this in ComfyUI and take advantage of the creativity that the nodes offer, right?

**Comment by thoddnn** [comment]
I've just coded some custom nodes that port DiffusionKit (similar to mflux) into ComfyUI. Currently, these are basic nodes for text-to-image workflows, but let me know which workflows you'd like to optimize. I'd be happy to help! 😊

ComfyUI-MLX: https://github.com/thoddnn/ComfyUI-MLX

**Reply by raysers** [reply]
> I've just coded some custom nodes that port DiffusionKit (similar to mflux) into ComfyUI. Currently, these are basic nodes for text-to-image workflows, but let me know which workflows you'd like to optimize. I'd be happy to help! 😊
> 
> ComfyUI-MLX: https://github.com/thoddnn/ComfyUI-MLX

Here is a convenient way to use Mflux on ComfyUI. 

https://github.com/raysers/Mflux-ComfyUI

However, I have only done a similar port of the WebUI and haven't really captured the essence of ComfyUI's design, which is to break it down into basic nodes so that users can understand the underlying logic of the workflow. My coding ability is limited, and I would greatly appreciate any help you can provide.
